spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java

```diff
@@ -183,44 +181,141 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
 		if (method == null) {
 			return invocation.proceed();
 		}
+
 		Method mostSpecificMethod = AopUtils
-				.getMostSpecificMethod(method, invocation.getThis().getClass());
+				.getMostSpecificMethod(invocation.getMethod(), invocation.getThis().getClass());
 		NewSpan newSpan = SleuthAnnotationUtils.findAnnotation(mostSpecificMethod, NewSpan.class);
 		ContinueSpan continueSpan = SleuthAnnotationUtils.findAnnotation(mostSpecificMethod, ContinueSpan.class);
 		if (newSpan == null && continueSpan == null) {
 			return invocation.proceed();
 		}
-		Span span = tracer().currentSpan();
-		if (newSpan != null || span == null) {
-			span = tracer().nextSpan().start();
+
+		Class<?> returnType = method.getReturnType();
```

Comment:
nit: returnType isn't used later, you can inline as method.getReturnType isn't clarified with the variable name

Reply:
inlined